armi/utils/gridEditor.py

```diff
@@ -0,0 +1,1783 @@
+        self.clicker.mode = mode
+
+        # make sure that gui elements that have to do with mode setting are consistent
+        if mode == GridGui.Mode.isPosition:
+            self.assemblyPalette.setActiveAssem(None)
+        self.controls.labelMode.SetSelection(mode)
+
+    def expandToFullCore(self, event):
+        self.clicker.growToFullCore()
+
+    def objectClicked(self, _idx):
+        """
+        Notify relevant controls that the object at the passed indices has been
+        activated.
+
+        This is needed to make the auto-increment stuff work in the fuel path editor.
+        Without some sort of event that provides a positive assertion that the user is
+        trying to interact with the layout, we can't know when to increment.
+        """
+        self.assemblyPalette.editorClicked()
+
+    def save(self, stream=None, full=False):
+        """
+        Save the blueprints to the passed stream, if provided. Otherwise prompt for a
+        file to save to.
+
+        This can save either the entire blueprints, or just the `grids:` section of the
+        blueprints, based on the passed ``full`` argument. Saving just the grid
+        blueprints can be useful when cobbling blueprints together with !include flags.
+        """
+        if stream is None:
+            # Prompt the user for a file name, open it, and call ourself again with that
+            # as the stream argument
+            if self._fName is None:
+                wd = os.getcwd()
+            else:
+                wd = os.path.split(self._fName)[0]
+
+            # Don't use the blueprints filename as the default if we are only saving the
+            # grids section; doing so may encourage users to overwrite their main
+            # blueprints file.
+            if full:
+                fName = self._fName or ""
+            else:
+                fName = ""
+
+            title = "Save blueprints to..." if full else "Save grid designs to..."
+
+            dlg = wx.FileDialog(
+                self,
+                message=title,
+                defaultDir=wd,
+                defaultFile=fName,
+                wildcard=self._wildcard,
+                style=wx.FD_SAVE | wx.FD_OVERWRITE_PROMPT,
+            )
+
+            if dlg.ShowModal() == wx.ID_OK:
+                path = dlg.GetPath()
+            else:
+                return
+
+            # Disallow overwriting the main blueprints with the grids section
+            if (
+                not full
+                and pathlib.Path(path).exists()
+                and pathlib.Path(path).samefile(self._fName)
+            ):
+                message = (
+                    "The chosen path, `{}` is the same as the main blueprints "
+                    "file. Try again with a different name.".format(path)
+                )
+
+                with wx.MessageDialog(
+                    self,
+                    message,
+                    "Overwriting top-level blueprints!",
```

Comment:
I think a lot of people will regularly hit this so we should expand the message more to say that we can't write blueprints but we can write subsections which can be included. (I don't think we should ask people to copy/paste though) more similar to what's said in the help msg. In my first encounter with an earlier version of this I overwrote my blueprints because I didn't quite understand the short message.

Reply:
to be fair, its supposed to make this message, then *not* save to that file. it was bugged, so it saved anyway. now it wont save over the original blueprints. Ive expanded the message to make it more clear